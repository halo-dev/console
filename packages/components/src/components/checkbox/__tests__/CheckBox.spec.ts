import { describe, expect, it } from "vitest";
import { VCheckbox } from "../index";
import { mount } from "@vue/test-utils";

describe("CheckBox", () => {
  it("should render", () => {
    expect(VCheckbox).toBeDefined();
    expect(mount(VCheckbox).html()).toMatchSnapshot();
  });

  it("should work with v-model:checked", async function () {
    const wrapper = mount({
      data() {
        return {
          checked: false,
        };
      },
      template: `
        <v-checkbox v-model:checked="checked"/>
      `,
      components: {
        VCheckbox,
      },
    });

    expect(wrapper.find("input").element.checked).toBe(false);
    expect(wrapper.findComponent(VCheckbox).classes()).not.toContain(
      "checkbox-wrapper-checked"
    );

    // change checked value
    await wrapper.setData({ checked: true });
    expect(wrapper.find("input").element.checked).toBe(true);
    expect(wrapper.findComponent(VCheckbox).classes()).toContain(
      "checkbox-wrapper-checked"
    );

    // click on checkbox
    await wrapper.find("input").setValue(false);
    expect(wrapper.vm.checked).toBe(false);
    expect(wrapper.find("input").element.checked).toBe(false);
    expect(wrapper.findComponent(VCheckbox).classes()).not.toContain(
      "checkbox-wrapper-checked"
    );
  });

  it("should work with label prop", async function () {
    const wrapper = mount(VCheckbox, {
      props: {
        checked: true,
      },
    });

    expect(wrapper.html()).not.toContain("label");

    await wrapper.setProps({ label: "label" });

    expect(wrapper.html()).toContain("label");
    expect(wrapper.find("label").text()).toBe("label");
  });
});
