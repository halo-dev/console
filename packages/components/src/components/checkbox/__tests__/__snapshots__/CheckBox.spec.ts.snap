// Vitest Snapshot v1

exports[`CheckBox > should render 1`] = `
"<div class=\\"checkbox-wrapper\\">
  <div class=\\"checkbox-inner\\"><input id=\\"checkbox\\" type=\\"checkbox\\"></div>
  <!--v-if-->
</div>"
`;
